src/controller/CommissioningWindowOpener.h

```diff
@@ -107,6 +108,45 @@ class CommissioningWindowOpener
                                        Callback::Callback<OnOpenCommissioningWindow> * callback, SetupPayload & payload,
                                        bool readVIDPIDAttributes = false);
 
+    /**
+     * @brief
+     *   Try to look up the device attached to our controller with the given
+     *   node id and ask it to re-enter commissioning mode with a PASE verifier
+     *   derived from the given information and the given discriminator. The
+     *   device will exit commissioning mode after a successful commissioning,
+     *   or after the given `timeout` time.
+     *
+     * @param[in] params        The parameters required to open an enhanced commissioning window
+     *                          with the provided or generated passcode.
+     * @param[in] callback      The function to be called on success or failure of opening the
+     *                          commissioning window. This will include the SetupPayload
+     *                          generated from provided parameters.
+     * @param[out] payload      The setup payload, not including the VID/PID bits,
+     *                          even if those were asked for, that is generated
+     *                          based on the passed-in information.  The payload
+     *                          provided to the callback function, unlike this
+     *                          out parameter, will include the VID/PID bits if
+     *                          readVIDPIDAttributes is true.
+     */
+    CHIP_ERROR OpenCommissioningWindow(const CommissioningWindowPasscodeParams & params,
+                                       Callback::Callback<OnOpenCommissioningWindow> * callback, SetupPayload & payload);
```

Comment:
callback should be part of `params`

Reply:
Moved the callback to params in https://github.com/project-chip/connectedhomeip/pull/34209/commits/8862cacfbc9519b71496b65ff923ddbd06cb31dc